### PR TITLE
Generate markdown files of rdoc from kpeg

### PR DESCRIPTION
**Pull request by hsbt**
I get conflict of backport from https://github.com/ruby/rdoc/pull/1096 changes. Should we remove generated code from kpeg?

### CODE REVIEW

File: common.mk [hunk 1]
```diff
@@ -1390,6 +1390,24 @@ $(srcdir)/ext/etc/constdefs.h: $(srcdir)/ext/etc/depend
 	$(exec) $(MAKE) -f - $(mflags) \
 		Q=$(Q) ECHO=$(ECHO) top_srcdir=../.. srcdir=. VPATH=../.. RUBY="$(BASERUBY)"
 
+KPEG= .bundle/bin/kpeg
```
Comment (kateinoigakukun) [inline common.mk:1393]:
```suggestion
KPEG = $(srcdir)/.bundle/bin/kpeg
```

File: common.mk [hunk 2]
```diff
@@ -1390,6 +1390,24 @@ $(srcdir)/ext/etc/constdefs.h: $(srcdir)/ext/etc/depend
...
+$(KPEG):
+	$(XRUBY) -C "$(srcdir)" bin/gem install --no-document \
+		--install-dir .bundle --conservative --silent "kpeg"
+
+generate-rdoc: $(srcdir)/lib/rdoc/markdown.rb $(srcdir)/lib/rdoc/markdown/literals.rb
+
+$(srcdir)/lib/rdoc/markdown.rb: $(srcdir)/lib/rdoc/markdown.kpeg $(KPEG)
+	$(ECHO) generating $@
+	$(XRUBY) -C "$(srcdir)" $(KPEG) -fs -o $(srcdir)/lib/rdoc/markdown.rb \
+		$(srcdir)/lib/rdoc/markdown.kpeg
```
Comment (kateinoigakukun) [inline common.mk:1404]: To repair out-of-source build
```suggestion
	$(XRUBY) -C "$(srcdir)" $(KPEG) -fs -o lib/rdoc/markdown.rb \
		lib/rdoc/markdown.kpeg
```

File: common.mk [hunk 3]
```diff
@@ -1390,6 +1390,24 @@ $(srcdir)/ext/etc/constdefs.h: $(srcdir)/ext/etc/depend
...
+
+$(srcdir)/lib/rdoc/markdown/literals.rb: $(srcdir)/lib/rdoc/markdown/literals.kpeg $(KPEG)
+	$(ECHO) generating $@
+	$(XRUBY) -C "$(srcdir)" .bundle/bin/kpeg -fs -o $(srcdir)/lib/rdoc/markdown/literals.rb \
+		$(srcdir)/lib/rdoc/markdown/literals.kpeg
```
Comment (kateinoigakukun) [inline common.mk:1409]: Ditto
```suggestion
	$(XRUBY) -C "$(srcdir)" .bundle/bin/kpeg -fs -o lib/rdoc/markdown/literals.rb \
		lib/rdoc/markdown/literals.kpeg
```

File: common.mk [hunk 4]
```diff
@@ -1390,6 +1390,22 @@ $(srcdir)/ext/etc/constdefs.h: $(srcdir)/ext/etc/depend
...
+$(KPEG): main
+$(KPEG):
+	$(XRUBY) -C "$(srcdir)" bin/gem install --no-document \
+		--install-dir .bundle --conservative --silent "kpeg"
+
+generate-rdoc: $(srcdir)/lib/rdoc/markdown.rb $(srcdir)/lib/rdoc/markdown/literals.rb
+
+$(srcdir)/lib/rdoc/markdown.rb: $(srcdir)/lib/rdoc/markdown.kpeg $(KPEG)
+	$(ECHO) generating $@
+	$(XRUBY) -C "$(srcdir)" $(KPEG) -fs -o lib/rdoc/markdown.rb lib/rdoc/markdown.kpeg
```
Comment (nobu) [inline common.mk:1403]: `KPEG` is now `$(srcdir)`-prefixed.
Here should be `.bundle/bin/kpeg`, or the prefix needs to be moved from `KPEG` definition to the dependency.
```suggestion
	$(XRUBY) -C "$(srcdir)" .bundle/bin/kpeg -fs -o lib/rdoc/markdown.rb lib/rdoc/markdown.kpeg
```

**Comment by hsbt** [comment]
I have a plan to extract RDoc code from `ruby/ruby` repo at Ruby 3.5.

https://github.com/ruby/ruby/pull/10282

We don't need to replace this.

**Pull request closed without merging**